### PR TITLE
Add a method for converting structs -> maps when bridging MM -> TF

### PR DESCRIPTION
**Pull request by rileykarson**
fyi @rambleraptor 

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by rileykarson** [comment]
We can merge this forward once we've reverted the original change & released, I tested a couple tests locally and they passed but I don't have the project quota to burn for a full run.

**Comment by rambleraptor** [comment]
thanks for the heads up!

File: third_party/terraform/resources/resource_google_project.go [hunk 1]
```diff
@@ -227,7 +227,12 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 	d.SetId(pid)
 
 	// Wait for the operation to complete
-	waitErr := resourceManagerOperationWait(config.clientResourceManager, op, "project to create")
+	opV1, waitErr := ConvertToMap(op)
+	if waitErr != nil {
```
Comment (chrisst) [inline third_party/terraform/resources/resource_google_project.go]: nit: this is a conversion error not really a wait error.

File: third_party/terraform/utils/convert.go [hunk 2]
```diff
@@ -28,6 +28,26 @@ func Convert(item, out interface{}) error {
 	return nil
 }
 
+// When converting to a map, we can't use setOmittedFields because FieldByName
+// fails. Luckily, we don't use the omitted fields anymore with generated
+// resources, and this function is used to bridge from handwritten -> generated.
+// Since this is a known type, we can create it inline instead of needing to
+// pass an object in.
+func ConvertToMap(item interface{}) (map[string]interface{}, error) {
```
Comment (chrisst) [inline third_party/terraform/utils/convert.go]: @nstogner looks like you'll be pulling this in soon and won't have to use https://github.com/GoogleCloudPlatform/magic-modules/pull/1496/files#diff-218e98d5cf92f706a23503eda8bbbdfdR10 any more.

**Pull request merged**